src/rabbit_exchange_type_consistent_hash.erl

```diff
@@ -84,9 +85,34 @@ route(#exchange { name      = Name,
             Destinations
     end.
 
-validate(_X) -> ok.
-
-validate_binding(_X, _B) -> ok.
+validate(#exchange { arguments = Args }) ->
+  case hash_args(Args) of
+    {undefined, undefined} -> ok;
+    {undefined, {_Type, Value}} ->
+      case lists:member(Value, ?PROPERTIES) of
+        true  -> ok;
+        false ->
+          rabbit_misc:protocol_error(precondition_failed,
+                                     "Unsupported property: ~s",
+                                     [Value])
+      end;
+    {_, undefined} -> ok;
+    {_, _} ->
+      rabbit_misc:protocol_error(precondition_failed,
+                                 "hash-header and hash-property are mutually exclusive",
```

Comment:
do you think you could add a test for the failure cases like here: https://github.com/rabbitmq/rabbitmq-delayed-message-exchange/blob/master/test/src/rabbit_exchange_type_delayed_message_test.erl#L33 ?